src/always-be-playing.js

```diff
@@ -135,6 +121,71 @@ export default class GapSkipper {
   }
 
   /**
+   * Handler for `waiting` events from the player
+   *
+   * @private
+   */
+  waiting_() {
+    let seekable = this.seekable();
+    let currentTime = this.tech_.currentTime();
+    let playlist = this.playlist();
+
+    if (this.tech_.seeking() ||
+        this.timer_ !== null ||
+        this.seekedAtTime === currentTime) {
+      return;
+    }
+
+    // check to see if we fell out of the live window
+    if (playlist &&
+        !playlist.endList &&
+        seekable.length &&
+        currentTime < seekable.start(0)) {
+      this.seekedAtTime = currentTime;
```

Comment:
It could be a relic of some old changes, but I remembered seeing waiting fired twice. I will switch it out and see if I can uncover why that was happening instead.